src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -233,6 +251,51 @@ public List<User> getAllUsersByGoogleId(String googleId) {
         return usersDb.getAllUsersByGoogleId(googleId);
     }
 
+    /**
+     * Deletes a student cascade its associated feedback responses, deadline extensions and comments.
+     *
+     * <p>Fails silently if the student does not exist.
+     */
+    public void deleteStudentCascade(String courseId, String studentEmail) {
+        Student student = getStudentForEmail(courseId, studentEmail);
+
+        if (student == null) {
+            return;
+        }
+
+        feedbackResponsesLogic
+                .deleteFeedbackResponsesInvolvedEntityOfCourseCascade(courseId, studentEmail);
+
+        if (usersDb.getStudentCountForTeam(student.getTeam().getName(), student.getCourseId()) == 1) {
+            // the student is the only student in the team, delete responses related to the team
+            feedbackResponsesLogic
+                    .deleteFeedbackResponsesInvolvedEntityOfCourseCascade(
+                        student.getCourse().getId(), student.getTeam().getName());
+        }
+
+        usersDb.deleteUser(student);
+        feedbackSessionsLogic.deleteFeedbackSessionsDeadlinesForUser(courseId, studentEmail);
+
+        updateStudentResponsesAfterDeletion(courseId);
+    }
+
+    private void updateStudentResponsesAfterDeletion(String courseId) {
+        feedbackResponsesLogic.updateFeedbackResponsesForDeletingStudent(courseId);
+    }
```

Comment:
This private method is not required. Extra level of indirection doesn't really add much value here, especially since this is a private method and there's no ambiguity in the original method name anyway.